src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -2675,9 +2675,9 @@ public void M()
             // (32,13): error CS0165: Use of unassigned local variable 'rest4'
             //             rest4.ToString(); // 8, 9
             Diagnostic(ErrorCode.ERR_UseDefViolation, "rest4").WithArguments("rest4").WithLocation(32, 13),
-            // (36,13): warning CS8602: Dereference of a possibly null reference.
-            //             rest5.ToString(); // 10
-            Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "rest5").WithLocation(36, 13)
+            // (34,33): warning CS8978: The annotation on the output type 'T' is ignored.
+            //         if (new C<T>() is [1, ..var rest5]) // ignored 10
```

Comment:
probably should check the annotation instead of state but the original warning still can't be avoided e.g. with `{} rest`.
what is the expected behavior here?

Reply:
(`T` is oblivious, I think `where T : notnull` should make it go away.)

Reply:
@jcouv thoughts on what diag message would make sense here?